controllers/helper.go

```diff
@@ -39,7 +44,91 @@ const (
 	trueString                            = "true"
 )
 
-var namespace = os.Getenv("NAMESPACE")
+type DrainAnnotationPredicate struct {
+	predicate.Funcs
+}
+
+func (DrainAnnotationPredicate) Create(e event.CreateEvent) bool {
+	logger := log.FromContext(context.TODO())
+	if e.Object == nil {
+		logger.Info("Create event: node has no drain annotation", "node", e.Object.GetName())
```

Comment:
this log lies. In general, I would avoid logging in these predicate functions.

Same applies for L59, L68, L72 and for `DrainStateAnnotationPredicate`